src/elements/input/src/index.tsx

```diff
@@ -26,6 +26,40 @@ export interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
   width?: Width
 }
 
+// On Android, there is a behaviour where the keyboard may sometimes open ON TOP of a focused input.
+// This effect fixes the problem by calling scrollIntoView on a window resize event - this event
+// is fired when the keyboard opens and pushes up the viewport.
+//
+// Related: https://stackoverflow.com/questions/23757345/android-does-not-correctly-scroll-on-input-focus-if-not-body-element?noredirect=1&lq=1
+const useScrollIntoView = (
+  inputRef: React.RefObject<HTMLInputElement>,
+  hasFocus: boolean
+) => {
+  const [isResizing, setIsResizing] = useState(false)
+
+  useEffect(() => {
+    const handleResize = () => setIsResizing(true)
+    window.addEventListener('resize', handleResize)
```

Comment:
Would it make sense to `debounce` the event listener so that it doesn't trigger multiple times during resizes?
That way if a user is manually resizing the page and the input is focused, we only trigger the scroll when the user stops resizing the browser

Reply:
Very good point, I've added a debounce around the resize handler.